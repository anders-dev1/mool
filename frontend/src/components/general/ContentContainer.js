import styled, {withTheme} from "styled-components";

/*
    Main content container div used everywhere on the website.
 */
const ContentContainer = styled.div`
  padding: 0.5em;
  background-color: white;
  border-radius: 5px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, .1), 0 8px 16px rgba(0, 0, 0, .1);
`

export default withTheme(ContentContainer);