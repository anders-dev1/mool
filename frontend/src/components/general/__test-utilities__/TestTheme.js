export const testTheme = {
    textColor: "black",
    primaryColor: "#007BFF",
    primaryHover: "rgba(0,123,255,0.33)",
    successColor: "#28A745",
    errorColor: "#F02849",
    backgroundColor: "#F0F2F5",
    buttonTextColor: "#FFFFFF",
    dividerColor: "#C7CBD1",
    inputColor: "#99A5B0",
    spinnerColor: "#FFFFFF transparent transparent transparent",
    font: "Acumin Pro, Arial, sans-serif !important;"
};