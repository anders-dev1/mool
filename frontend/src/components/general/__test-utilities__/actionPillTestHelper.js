import {screen} from "@testing-library/react";

export function findActionPillByIcon(icon) {
    return screen.getByText(icon).closest('div');
}

export function findActionPillNumberByIcon(icon) {
    const pill = findActionPillByIcon(icon);
    const numberP = pill.lastChild.lastChild;
    return numberP.textContent;
}