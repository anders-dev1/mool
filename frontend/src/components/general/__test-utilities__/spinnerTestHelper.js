export function findSpinner(container){
    return container.getElementsByClassName('lds-ring')[0];
}