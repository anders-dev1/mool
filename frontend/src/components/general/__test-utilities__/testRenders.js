import React from 'react'
import {render} from '@testing-library/react'
import {Provider} from 'react-redux'
import {setupStore, store} from "../../../redux/store";
import {testTheme} from "./TestTheme";
import {ThemeProvider} from "styled-components";

// This is used to set up the store for tests.
export function renderWithProviders(
    ui,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        testStore = setupStore(preloadedState),
        ...renderOptions
    } = {}
) {
    function Wrapper({children}) {
        return(
            <Provider store={testStore}>
                <ThemeProvider theme={testTheme}>
                {children}
                </ThemeProvider>
            </Provider>
        );
    }

    return {testStore, ...render(ui, {wrapper: Wrapper, ...renderOptions})}
}

// This is used with the react-test-renderer to render dom in tests.
export function internalsForRenderer(
    ui,
    {
        preloadedState = {},
        testStore = setupStore(preloadedState)
    } = {}
){
    function Wrapper(children) {
        return(
            <Provider store={testStore}>
                <ThemeProvider theme={testTheme}>
                    {children}
                </ThemeProvider>
            </Provider>
        );
    }

    return Wrapper(ui);
}