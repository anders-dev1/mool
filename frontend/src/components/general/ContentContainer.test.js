import renderer from "react-test-renderer";
import ContentContainer from "./ContentContainer";
import {testTheme} from "./__test-utilities__/TestTheme";
import {ThemeProvider} from "styled-components";

test('renders correctly', () => {
   const tree = renderer.create(
       <ThemeProvider theme={testTheme}>
         <ContentContainer>This is a test</ContentContainer>
       </ThemeProvider>
   ).toJSON();
   expect(tree).toMatchSnapshot();
});