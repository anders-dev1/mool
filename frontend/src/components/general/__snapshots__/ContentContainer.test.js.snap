// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bczRLJ gvFepU"
>
  This is a test
</div>
`;
